### PR TITLE
Suggest changing type to const parameters if we encounter a type in the trait bound position

### PR DESCRIPTION
**Pull request by fmease**
The first commit is just drive-by cleanup.

Provide a structured suggestion if the user forgot to prefix a “const parameter” with `const`, e.g., in `struct Tagged<TAG: u64>;`. This happens to me from time to time. Maybe C++ devs are also prone to this mistake given template syntax looks like `template<typename T, uint32_t N>`.

### CODE REVIEW

**Comment by fmease** [comment]
r? compiler (b-naber has been inactive for more than two months)

File: compiler/rustc_resolve/src/late/diagnostics.rs [hunk 1]
```diff
@@ -1136,6 +1139,55 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
...
+        // We don't include `DefKind::Str` and `DefKind::AssocTy` as they can't be reached here anyway.
+        let applicability = match res {
+            Some(Res::PrimTy(PrimTy::Int(_) | PrimTy::Uint(_) | PrimTy::Bool | PrimTy::Char)) => {
+                Applicability::MachineApplicable
+            }
+            // FIXME(const_generics): Add `DefKind::TyParam` and `SelfTyParam` once we support generic
+            // const generics. Of course, `Struct` and `Enum` may contain ty params, too, but the
+            // benefits of including them here outweighs the small number of false positives.
+            Some(Res::Def(DefKind::Struct | DefKind::Enum, _))
+                if self.r.tcx.features().adt_const_params =>
```
Comment (compiler-errors) [inline compiler/rustc_resolve/src/late/diagnostics.rs:1160]: ❤️

**Review by compiler-errors — APPROVED** [review]
very cool

**Pull request merged**